bubuku/features/restart_if_dead.py

```diff
@@ -8,48 +8,47 @@
 
 _LOG = logging.getLogger('bubuku.features.restart_if_dead')
 
-
 class CheckBrokerStopped(Check):
     def __init__(self, broker: BrokerManager, zk: BukuExhibitor):
         super().__init__()
         self.broker = broker
         self.zk = zk
         self.need_check = True
-        self.last_zk_session_check = datetime.now()
+        self.last_zk_session_failed_check = None
 
     def check(self) -> Change:
         if not self.need_check:
             return None
-        if self.is_running_and_registered(should_retry=True):
+        if not self.should_restart():
             return None
 
         _LOG.warning('Oops! Broker is dead, triggering restart')
         self.need_check = False
 
         # Do not start if broker is running and registered
         def _cancel_if():
-            return self.is_running_and_registered()
+            return not self.broker.is_running() or self.broker.is_registered_in_zookeeper()
```

Comment:
Seems that the the line meaning has changed: 
was - cancel if broker is running and registered
now - cancel if broker is (not running) or registered

Reply:
Oops

Reply:
It should bail reboot when process is running and registered in zookeeper